src/Widgets/PartitionMenu.vala

```diff
@@ -0,0 +1,193 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2016-2018 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Michael Aaron Murphy <michael@system76.com>
+ */
+
+public delegate void SetMount (Installer.Mount mount);
+
+public delegate void UnsetMount (string partition);
+
+public class Installer.PartitionMenu : Gtk.Popover {
+    private Gtk.Grid grid;
+    private Gtk.Switch use_partition;
+    private Gtk.ComboBoxText use_as;
+    private Gtk.Entry custom;
+    private Gtk.ComboBoxText type;
+    private string partition_path;
+    private string parent_disk;
+
+    public PartitionMenu (string path, string parent, SetMount set_mount, UnsetMount unset_mount) {
+        partition_path = path;
+        parent_disk = parent;
+        grid = new Gtk.Grid ();
```

Comment:
Let's keep all the grid properties and methods together, so this should go down in front of line 52